client/components/search/index.jsx

```diff
@@ -271,19 +280,26 @@ const Search = React.createClass( {
 			spellCheck: 'false'
 		};
 
-		searchClass = classNames( this.props.additionalClasses, {
+		const searchClass = classNames( this.props.additionalClasses, this.props.dir, {
 			'is-expanded-to-container': this.props.fitsContainer,
 			'is-open': isOpenUnpinnedOrQueried,
 			'is-searching': this.props.searching,
-			rtl: this.props.dir === 'rtl',
-			ltr: this.props.dir === 'ltr',
+			'no-close-button': this.props.hideClose,
```

Comment:
Since an accessor to clear fields isn't something the users expect (with probably the exception of some fields on mobile) I opted to have a clearer UI for the combination of search + pill selector. Clearing is done either by deleting the text or reloading the page.

I still think that `×` should be a default on tho.